packages/block-library/src/image/view.js

```diff
@@ -187,6 +196,57 @@ const { state, actions, callbacks } = store(
 					}
 				}
 			},
+			preloadImage() {
+				const { imageId } = getContext();
+				const imageMetadata = state.metadata[ imageId ];
+				const uploadedSrc = imageMetadata.uploadedSrc;
+
+				// Bails if it has already been preloaded. This could help
+				// prevent unnecessary preloading of the same image multiple times,
+				// leading to duplicate link elements in the document head.
+				if ( state.preloadedImageIds.has( imageId ) ) {
+					return;
+				}
+
+				const imageLink = document.createElement( 'link' );
+				imageLink.rel = 'preload';
+				imageLink.as = 'image';
+				imageLink.href = uploadedSrc;
+
+				// Apply srcset if available for responsive preloading
+				const srcset = imageMetadata.lightboxSrcset;
+				if ( srcset ) {
+					imageLink.setAttribute( 'imagesrcset', srcset );
+					imageLink.setAttribute( 'imagesizes', '100vw' );
+				}
+
+				document.head.appendChild( imageLink );
+				state.preloadedImageIds.add( imageId );
+			},
+			preloadImageWithDelay() {
+				const { imageId } = getContext();
+
+				// Cancels any previous preload timer for the same image.
+				if ( state.preloadTimers && state.preloadTimers[ imageId ] ) {
+					clearTimeout( state.preloadTimers[ imageId ] );
+				}
+
+				// Set a new timer to preload the image after a short delay.
+				state.preloadTimers[ imageId ] = setTimeout(
+					withScope( () => {
+						actions.preloadImage();
+						delete state.preloadTimers[ imageId ];
+					} ),
+					200
+				);
+			},
+			cancelPrefetch() {
+				const { imageId } = getContext();
+				if ( state.preloadTimers && state.preloadTimers[ imageId ] ) {
+					clearTimeout( state.preloadTimers[ imageId ] );
+					delete state.preloadTimers[ imageId ];
+				}
+			},
```

Comment:
Per above, if using `Map`:

```suggestion
			preloadImageWithDelay() {
				const { imageId } = getContext();

				// Cancels any previous preload timer for the same image.
				if ( state.preloadTimers.has( imageId ) ) {
					clearTimeout( state.preloadTimers.get( imageId ) );
				}

				// Set a new timer to preload the image after a short delay.
				state.preloadTimers.set( imageId, setTimeout(
					withScope( () => {
						actions.preloadImage();
						state.preloadTimers.delete( imageId );
					} ),
					200
				) );
			},
			cancelPrefetch() {
				const { imageId } = getContext();
				if ( state.preloadTimers.has( imageId ) ) {
					clearTimeout( state.preloadTimers.get( imageId ) );
					state.preloadTimers.delete( imageId );
				}
			},
```

Reply:
Thanks for the suggestion! I’ve updated the code to use `Map` accordingly and simplified the timer management as you proposed.